ichnaea/webapp/app.py

```diff
@@ -51,20 +49,37 @@ def worker_exit(server, worker):
     shutdown_worker(_APP)
 
 
-def log_access_factory(wsgi_app):
-    """WSGI middleware for logging HTTP requests."""
+def log_middleware(wsgi_app):
+    """WSGI middleware for logging."""
 
     def handle(environ, start_response):
         method = environ["REQUEST_METHOD"]
         path = environ.get("PATH_INFO", "")
+        start = time.time()
+        structlog.threadlocal.clear_threadlocal()
+        structlog.threadlocal.bind_threadlocal(http_method=method, http_path=path)
 
         def log_response(status, headers, exc_info=None):
-            content_length = ""
+            duration = time.time() - start
+            try:
+                status_code = int(status.split()[0])
+            except (ValueError, AttributeError, IndexError):
+                status_code = status
+
+            params = {
+                "http_status": status_code,
+                "duration": round(duration, 3),  # Round to milliseconds
+            }
+
+            content_length_str = ""
             for key, val in headers:
                 if key.lower() == "content-length":
-                    content_length = "(%s)" % val
+                    params["content_length"] = val
+                    content_length_str = f" ({val})"
                     break
-            LOGGER.info("%s %s - %s %s", method, path, status, content_length)
+
+            logger = structlog.get_logger("canonical-log-line")
```

Comment:
Here's where the logger name changed from `ichnaea` to `canonical-log-line`. I'm 50% / 50% if this is the right thing to do.